component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java

```diff
@@ -43,31 +43,28 @@ protected DefaultResponseMetricEventBuilder() {
 
     @Override
     public boolean validate() throws MetricReportingException {
-        for (String attributeKey : requiredAttributes) {
-            Object attribute = eventMap.get(attributeKey);
+        for (Map.Entry<String, Class> entry : requiredAttributes.entrySet()) {
+            Object attribute = eventMap.get(entry.getKey());
             if (attribute == null) {
-                throw new MetricReportingException(attributeKey + " is missing in metric data");
-            } else if (attribute instanceof String) {
-                if (((String) attribute).isEmpty()) {
-                    throw new MetricReportingException(attributeKey + " is missing in metric data");
-                }
+                throw new MetricReportingException(entry.getKey() + " is missing in metric data. This metric event "
+                                                           + "will not be processed further.");
+            } else if (!attribute.getClass().equals(entry.getValue())) {
+                throw new MetricReportingException(entry.getKey() + " is expecting a " + entry.getValue() + " "
+                                                           + "type attribute while attribute of type "
+                                                           + attribute.getClass() + " is present.");
```

Comment:
this also we can move to a single line I think.

Reply:
to do that have to drag it forward of the opening brace which is not compliant with code styles template